diff --git a/main.go b/main.go
index 4642335e..b77f3742 100644
--- a/main.go
+++ b/main.go
@@ -99,6 +99,39 @@ func main() {
 			Name:  "systemd-cgroup",
 			Usage: "enable systemd cgroup support, expects cgroupsPath to be of form \"slice:prefix:name\" for e.g. \"system.slice:runc:434234\"",
 		},
+		cli.IntFlag{
+			Name:  "cpu",
+			Value: 1,
+			Usage: "default number of CPUs of guest VM",
+		},
+		cli.IntFlag{
+			Name:  "mem",
+			Value: 64,
+			Usage: "default RAM size of guest VM in MiB",
+		},
+		cli.StringFlag{
+			Name:  "dns",
+			Value: "",
+			Usage: "comma-separated list of DNS servers",
+		},
+		cli.StringFlag{
+			Name:  "dns-opts",
+			Value: "",
+			Usage: "DNS options to use",
+		},
+		cli.StringFlag{
+			Name:  "dns-search",
+			Value: "",
+			Usage: "DNS search domains to use",
+		},
+		cli.BoolFlag{
+			Name:  "sigusr",
+			Usage: "enable sigusr commands",
+		},
+		cli.BoolFlag{
+			Name:  "nestedvm",
+			Usage: "enable support for nested VMs",
+		},
 	}
 	app.Commands = []cli.Command{
 		checkpointCommand,
diff --git a/utils.go b/utils.go
index 8ed1a88e..bb83178c 100644
--- a/utils.go
+++ b/utils.go
@@ -64,6 +64,9 @@ func setupSpec(context *cli.Context) (*specs.Spec, error) {
 	if err != nil {
 		return nil, err
 	}
+	if err := turnToRunq(context, spec); err != nil {
+		return nil, err
+	}
 	return spec, nil
 }
 
