diff --git a/exec.go b/exec.go
index ae8e60e4..f89ddeeb 100644
--- a/exec.go
+++ b/exec.go
@@ -91,6 +91,7 @@ following will output a list of processes running in the container:
 		},
 	},
 	Action: func(context *cli.Context) error {
+		return fmt.Errorf("RunQ does not support 'docker exec'. Use 'runq-exec' instead.")
 		if err := checkArgs(context, 1, minArgs); err != nil {
 			return err
 		}
diff --git a/main.go b/main.go
index 072447d1..b8bdbc4c 100644
--- a/main.go
+++ b/main.go
@@ -112,6 +112,35 @@ func main() {
 			Value: "auto",
 			Usage: "ignore cgroup permission errors ('true', 'false', or 'auto')",
 		},
+		cli.IntFlag{
+			Name:  "cpu",
+			Value: 1,
+			Usage: "default number of CPUs of guest VM",
+		},
+		cli.IntFlag{
+			Name:  "mem",
+			Value: 64,
+			Usage: "default RAM size of guest VM in MiB",
+		},
+		cli.StringFlag{
+			Name:  "dns",
+			Value: "",
+			Usage: "comma-separated list of DNS servers",
+		},
+		cli.StringFlag{
+			Name:  "dns-opts",
+			Value: "",
+			Usage: "DNS options to use",
+		},
+		cli.StringFlag{
+			Name:  "dns-search",
+			Value: "",
+			Usage: "DNS search domains to use",
+		},
+		cli.BoolFlag{
+			Name:  "nestedvm",
+			Usage: "enable support for nested VMs",
+		},
 	}
 	app.Commands = []cli.Command{
 		checkpointCommand,
diff --git a/utils.go b/utils.go
index 5165336f..9aa0a97a 100644
--- a/utils.go
+++ b/utils.go
@@ -66,6 +66,9 @@ func setupSpec(context *cli.Context) (*specs.Spec, error) {
 	if err != nil {
 		return nil, err
 	}
+	if err := turnToRunq(context, spec); err != nil {
+		return nil, err
+	}
 	return spec, nil
 }
 
